src/redux/orderSlice.ts

```diff
@@ -0,0 +1,192 @@
+  order: {
+    priceBreakdown: {
+      offerPrice: number;
+      discountAmount: number;
+      taxValue: number;
+      customerServiceFee: number;
+      paymentMethodFee: number;
+    };
+    discount: { applied: boolean };
+    taxRate: number;
+    country: string;
+    currency: string;
+    totalPrice: number;
+    id: string;
+    couponDetails: object;
+  };
+  loading: boolean;
+  error: string | null;
+  couponDetails: {
+    showMessage: boolean;
+    message: string;
+    messageType: typeof MESSAGE_TYPE_SUCCESS | typeof MESSAGE_TYPE_FAIL;
+  };
+  isCouponLoading: boolean;
+  couponError: string | null;
+  isUpdateLoading: boolean;
+};
+
+const initialState: InitialState = {
+  order: {
+    priceBreakdown: {
+      offerPrice: 0,
+      discountAmount: 0,
+      taxValue: 0,
+      customerServiceFee: 0,
+      paymentMethodFee: 0
+    },
+    discount: { applied: false },
+    taxRate: 0,
+    country: '',
+    currency: '',
+    totalPrice: 0,
+    id: '',
+    couponDetails: {}
+  },
+  loading: true,
+  error: null,
+  couponDetails: {
+    showMessage: false,
+    message: '',
+    messageType: MESSAGE_TYPE_SUCCESS
+  },
+  isCouponLoading: false,
+  couponError: null,
+  isUpdateLoading: false
+};
+
+export const fetchCreateOrder = createAsyncThunk(
+  'order/createOrder',
+  async (offerId: string, { rejectWithValue }) => {
+    try {
+      const { order } = await createOrder(offerId);
+      return order;
+    } catch (err) {
+      if (isErrorMsg(err)) return rejectWithValue(err.message);
+      return err;
+    }
+  }
+);
+
+export const fetchUpdateOrder = createAsyncThunk(
+  'order/updateOrder',
+  async (
+    { id, payload }: { id: string; payload: { [key: string]: unknown } },
+    { rejectWithValue }
+  ) => {
+    try {
+      const { order } = await updateOrder(id, payload);
+      return order;
+    } catch (err) {
+      if (isErrorMsg(err)) return rejectWithValue(err.message);
+      return err;
+    }
+  }
+);
+
+export const fetchUpdateCoupon = createAsyncThunk(
+  'order/updateCoupon',
+  async (
+    { id, couponCode }: { id: string; couponCode: string },
+    { rejectWithValue }
+  ) => {
+    try {
+      const { order } = await updateOrder(id, couponCode);
+      return order;
+    } catch (err) {
+      if (isErrorMsg(err)) return rejectWithValue(err.message);
+      return err;
+    }
+  }
+);
+
+export const fetchGetOrder = createAsyncThunk(
+  'order/getOrder',
+  async (id: string, { rejectWithValue }) => {
+    try {
+      const { order } = await getOrder(id);
+      return order;
+    } catch (err) {
+      if (isErrorMsg(err)) return rejectWithValue(err.message);
+      return err;
+    }
+  }
+);
+
+export const orderSlice = createSlice({
+  name: 'order',
+  initialState,
+  reducers: {},
+  extraReducers: builder => {
+    builder.addCase(fetchCreateOrder.pending, state => {
+      state.loading = true;
+    });
+    builder.addCase(fetchCreateOrder.fulfilled, (state, { payload }) => {
```

Comment:
we could try to add PayloadAction if it is possible in extraReducers

Reply:
I added PayloadAction to fulfilled actions, but I couldn't find information how to properly add it to rejected ones.